### ISSUE TITLE
Frontier in Multisite

### ISSUE THREAD

**Issue by carlosrovira** [DavidCramer/pods-frontier#34]
I just found an issue with my testing of Frontier in Multisite.

I you "Network Activate" Frontier, the main site shows blank page. But other sub sites works ok.

To workaround this issue, I network deactivate, and activate in the sub site I need it, since newer sub sites will be created as a copy of the before mentioned sub site with Frontier activated.

For more info, I'm using a Gavick Theme (Gavern Framework) for my theme, but I tried switching to default WP theme and it show the same blank page issue.

Best

Carlos

**Comment by Shelob9** [comment]
I can confirm this with Pods and Pods-Frontier network activated main site gives this error:

`Fatal error: Call to undefined function frontier_get_regex() in ~/plugins/pods-frontier/pods-frontier.php on line 745`

Since `frontier_get_regex()` is in the Pods Template component so this must have to do with order of including files.

**Comment by DavidCramer** [comment]
I can confirm it to.
I'll take a look at the includes.

-----Original Message-----

**Comment by Pezzab** [comment]
I was about to submit what looked like a bug and came across this.  I came across the same issue as mentioned here (I also got the fatal error Josh mentions above), I'm on Multisite too.

It might be worth checking if Templates are activated in Components - I stopped getting the error once I sorted this.

Perhaps the error could be caught and a message reminding the user to activate templates could appear instead?

**Comment by Shelob9** [comment]
@Pezzab Have you tested what happens if you manually activate the Templates component before activating Frontier? Templates gets turned on with `pods_require_component()` which doesn't run until '[setup_theme](http://adambrown.info/p/wp_hooks/hook/setup_theme)', which is probably too late on network. I can't test right now, but I bet that's it.

If need be we can do something like this:
https://github.com/pods-framework/pods-code-library/blob/master/example/misc/examples/activate-pods-componets.php

**Comment by Pezzab** [comment]
As far as I can tell if Templates is activated first then there isn't an issue.

**Comment by Shelob9** [comment]
Perry, that's excellent news. Thanks for testing this. I know how to fix this then and will later tonight.

**Comment by Shelob9** [comment]
@Pezzab and @carlosrovira I think this latest commit will fix this. Can either of you confirm?

**Comment by Pezzab** [comment]
Hi Josh,

I've tried different combinations of deactivating/reactivating (via network and on a sub site) Pods Frontier and Pods Frontier Templates and it hasn't fallen over.

Do you reckon it needs trying out on a clean WP install just to be sure - I deleted all the podsrel tables and reinstalled Pods but some settings were carried over from before (components were still installed)?

**Comment by Shelob9** [comment]
@Pezzab  No, this should force the Template component to be activated when components are loaded. It works when not in multisite. Still trying to figure out why it doesn't when not in multisite.

Thanks for your testing.

**Comment by carlosrovira** [comment]
Hi Just tested with no luck. Sorry the problem persist :(

**Comment by Shelob9** [comment]
@Pezzab and @carlosrovira thanks for your help on this. My latest commit works on my end, but I'd appreciate if you could test as well.

**Comment by alexedev** [comment]
I got the same error but I solved it by updating PHP to higher version.

**Comment by Shelob9** [comment]
Might be time to make Frontier 5.3+ only. This plus the Template editor issue...
